backend/src/controllers/user.controller.ts

```diff
@@ -41,9 +41,17 @@ export const getOne = async (req: Request, res: Response): Promise<Response> =>
 };
 
 export const create = async (req: Request, res: Response): Promise<Response> => {
-  const user = req.body;
+  const body = req.body as IUser;
   try {
-    const result = await User.create(user);
+    if (body.email) {
+      const numberOfUsers = await User.findOne({ email: body.email }).countDocuments();
```

Comment:
`countDocuments()` is not required. We want to check if any record exists. When nothing found, numberOfUsers should be null